source/production/F0.Minesweeper.Logic/Minefield.cs

```diff
@@ -46,12 +46,12 @@ public IGameUpdateReport Uncover(Location location)
 
 			GameStatus gameStatus = GetGameStatus();
 
-			if(gameStatus != GameStatus.InProgress)
+			if (gameStatus != GameStatus.InProgress)
 			{
-				return new GameUpdateReport(gameStatus, minefield.ToArray());
+				return new GameUpdateReport(gameStatus, minefield);
```

Comment:
Since we hand off a reference to the member of this type, we should pass here a copy instead.